src/raft_log.rs

```diff
@@ -379,19 +379,24 @@ impl<T: Storage> RaftLog<T> {
     }
 
     /// Returns entries starting from a particular index and not exceeding a bytesize.
-    pub fn entries(&self, idx: u64, max_size: impl Into<Option<u64>>) -> Result<Vec<Entry>> {
+    pub fn entries(
+        &self,
+        idx: u64,
+        max_size: impl Into<Option<u64>>,
+        context: GetEntriesContext,
+    ) -> Result<Vec<Entry>> {
         let max_size = max_size.into();
         let last = self.last_index();
         if idx > last {
             return Ok(Vec::new());
         }
-        self.slice(idx, last + 1, max_size)
+        self.slice(idx, last + 1, max_size, context)
     }
 
-    /// Returns all the entries.
+    /// Returns all the entries. Only used by tests.
     pub fn all_entries(&self) -> Vec<Entry> {
```

Comment:
How about marking it as private?

Reply:
it's used in `harness/test` which is out of the crate. So can't make it private

Reply:
Then maybe can make it doc hidden.